forui/lib/src/widgets/dialog/dialog.dart

```diff
@@ -0,0 +1,193 @@
+  final Duration insetAnimationDuration;
+
+  /// The curve to use for the animation shown when the system keyboard intrudes
+  /// into the space that the dialog is placed in.
+  ///
+  /// Defaults to [Curves.decelerate].
+  final Curve insetAnimationCurve;
+
+  /// The semantic label of the dialog used by accessibility frameworks to announce screen transitions when the dialog
+  /// is opened and closed.
+  ///
+  /// See also:
+  ///  * [SemanticsConfiguration.namesRoute], for a description of how this
+  ///    value is used.
+  final String? semanticLabel;
+
+  /// The builder.
+  final Widget Function(BuildContext, FDialogStyle) builder;
+
+  /// Creates a [FDialog] with a title, subtitle, and possible actions.
+  FDialog({
+    required List<Widget> actions,
+    this.style,
+    this.insetAnimationDuration = const Duration(milliseconds: 100),
+    this.insetAnimationCurve = Curves.decelerate,
+    String? semanticLabel,
+    String? title,
+    String? subtitle,
+    FDialogAlignment alignment = FDialogAlignment.vertical,
+    super.key,
+  }):
+    semanticLabel = semanticLabel ?? title ?? subtitle,
+    builder = switch (alignment) {
+      FDialogAlignment.horizontal => (context, style) => FHorizontalDialogContent(
+        style: style.horizontal,
+        title: title,
+        subtitle: subtitle,
+        actions: actions,
+      ),
+      FDialogAlignment.vertical => (context, style) => FVerticalDialogContent(
+        style: style.vertical,
+        title: title,
+        subtitle: subtitle,
+        actions: actions,
+      ),
+    };
+
+  /// Creates a [FDialog] with a custom builder.
+  const FDialog.raw({
+    required this.builder,
+    this.style,
+    this.insetAnimationDuration = const Duration(milliseconds: 100),
+    this.insetAnimationCurve = Curves.decelerate,
+    this.semanticLabel,
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = context.theme;
+    final style = this.style ?? theme.dialogStyle;
+    final typography = theme.typography;
+
+    return AnimatedPadding(
+      padding: MediaQuery.viewInsetsOf(context) + style.insetPadding,
+      duration: insetAnimationDuration,
+      curve: insetAnimationCurve,
+      child: MediaQuery.removeViewInsets(
+        removeLeft: true,
+        removeTop: true,
+        removeRight: true,
+        removeBottom: true,
+        context: context,
+        child: Align(
+          child: DefaultTextStyle(
+            style: context.theme.typography.toTextStyle(
+              fontSize: typography.base,
+              color: context.theme.colorScheme.foreground,
+            ),
+            child: Semantics(
+              scopesRoute: true,
+              explicitChildNodes: true,
+              namesRoute: true,
+              label: semanticLabel,
+              child: ConstrainedBox(
+                constraints: const BoxConstraints(
+                  minWidth: 280.0,
+                  maxWidth: 560.0,
```

Comment:
Should we place this in styles instead of leaving them as magic values?

Reply:
I personally don't think these values should be configurable.

Reply:
Would it be possible if a content requires more than the max width?